### ISSUE TITLE
`this.req.destroyed = true` in abort does not abort requests

### ISSUE THREAD

**Issue by ulisesbocchio** [visionmedia/superagent#1741]
I just upgraded my codebase to latest version of superagent and noticed my unit tests breaking because of this:
https://github.com/visionmedia/superagent/blob/29fd1f917a6cc78c9a2a9984269c06f8b4e63dcb/src/request-base.js#L505-L508

When I remove that line, they pass. I'd like to know what exactly are you referring to in your comment. In Node 16 LTS `req.abort()` works as expected without having to set the flag manually. Also, the documentation now discourages the use of `req.abort()` in favor of `req.destroy()` which I tested and works great too. For instance you could do `req.destroy(new Error('Aborted'))`.
To give you an example of how setting that flag does't work as expected here's some reference code:
```js
const agentReq = superagent.get('http://localhost:5677/test');

const s = http.createServer((req, res) => {
    console.log(`<-- ${req.method} ${req.url}`);
    agentReq.abort();
    req.on('error', e => console.log(`XXX --> ${req.method} ${req.url} ${e.message}`));
    req.on('end', () => console.log(`--> ${req.method} ${req.url} 200`));
    setTimeout(() => {
        res.setHeader('Content-Type', 'application/json');
        res.writeHead(200);
        res.end('{ "test": "hello" }');
    }, 500);
});
s.listen(5677);

agentReq
    .then(
        r => console.log('Request Body:', r.body),
        e => console.log('Request Error:', e.message)
    )
    .finally(() => new Promise(resolve => s.close(resolve)));
```
Basically here there's a request going out to a server with a handler that's gonna abort that request while it's being handled. The handler has an "access logger" that logs out the result of the server side request, and the superagent request result is also logged
when aborting the superagent request  you'd expect this output:
```bash
<-- GET /test
Request Error: Aborted
XXX --> GET /test aborted
```
but instead you get:
```bash
<-- GET /test
Request Error: Aborted
--> GET /test 200
```
